x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/DataLifecycleServiceRuntimeSecurityIT.java

```diff
@@ -0,0 +1,274 @@
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return List.of(DataLifecyclePlugin.class, LocalStateSecurity.class, DataStreamsPlugin.class, SystemDataStreamTestPlugin.class);
+    }
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
+        Settings.Builder settings = Settings.builder().put(super.nodeSettings(nodeOrdinal, otherSettings));
+        settings.put(DataLifecycleService.DLM_POLL_INTERVAL, "1s");
+        settings.put(DataLifecycle.CLUSTER_DLM_DEFAULT_ROLLOVER_SETTING.getKey(), "min_docs=1,max_docs=1");
+        return settings.build();
+    }
+
+    public void testRolloverLifecycleAndForceMergeAuthorized() throws Exception {
+        String dataStreamName = "metrics-foo";
+        // empty lifecycle contains the default rollover
+        prepareDataStreamAndIndex(dataStreamName, new DataLifecycle());
+
+        assertBusy(() -> {
+            assertNoAuthzErrors();
+            List<Index> backingIndices = getDataStreamBackingIndices(dataStreamName);
+            assertThat(backingIndices.size(), equalTo(2));
+            String backingIndex = backingIndices.get(0).getName();
+            assertThat(backingIndex, backingIndexEqualTo(dataStreamName, 1));
+            String writeIndex = backingIndices.get(1).getName();
+            assertThat(writeIndex, backingIndexEqualTo(dataStreamName, 2));
+        });
+
+        // Index another doc to force another rollover and trigger an attempted force-merge. The force-merge may be a noop under
+        // the hood but for authz purposes this doesn't matter, it only matters that the force-merge API was called
+        indexDoc(dataStreamName);
+        assertBusy(() -> {
+            assertNoAuthzErrors();
+            List<Index> backingIndices = getDataStreamBackingIndices(dataStreamName);
+            assertThat(backingIndices.size(), equalTo(3));
+        });
+    }
+
+    public void testRolloverAndRetentionAuthorized() throws Exception {
+        String dataStreamName = "metrics-foo";
+        prepareDataStreamAndIndex(dataStreamName, new DataLifecycle(TimeValue.timeValueMillis(0)));
+
+        assertBusy(() -> {
+            assertNoAuthzErrors();
+            List<Index> backingIndices = getDataStreamBackingIndices(dataStreamName);
+            assertThat(backingIndices.size(), equalTo(1));
+            // we expect the data stream to have only one backing index, the write one, with generation 2
+            // as generation 1 would've been deleted by DLM given the lifecycle configuration
+            String writeIndex = backingIndices.get(0).getName();
+            assertThat(writeIndex, backingIndexEqualTo(dataStreamName, 2));
+        });
+    }
+
+    public void testUnauthorized() throws Exception {
+        // the DLM user is not authorized for this pattern since it matches the security index
+        String dataStreamName = ".security-foo";
```

Comment:
nit: use random values appended to the known prefix to help illustrate what actually is unauthorized (and randomness between the two unauthorized prefix)

Reply:
After changing the privileges for the internal user I realized that `".security-foo"` was only failing because of the name exclusion pattern we had previously, and that `".security-foo"` isn't actually treated as a restricted index (since that pattern is not registered as an actual system index). 

For this test to fail as expected, we actually need to use a correct and registered system index name. I've gone with using the security index main alias, without randomization and added a clarifying comment. I could alternatively try to get a list of registered system indices, but would prefer to just go with the security one as it's simpler and exercises the correct failure scenario. Let me know if you feel strongly though!

Reply:
> I've gone with using the security index main alias, without randomization and added a clarifying comment.

SGTM